internal/abci/utils.go

```diff
@@ -77,3 +79,54 @@ func nonceList(txns []*transactions.Transaction) []uint64 {
 	}
 	return nonces
 }
+
+// The default parameter updates *should* generally be nothing (and nil fields)
+// but we have this for historical reasons. This is a function rather than a
+// var to prevent accidental modifications.
+func defaultParamUpdates() consensus.ParamUpdates {
+	return consensus.ParamUpdates{
+		Version: &chain.VersionParams{}, // App = 0, every time...
+		Validator: &chain.ValidatorParams{
+			PubKeyTypes: []string{"ed25519"},
```

Comment:
I already noted elsewhere, but what is the reasoning for forcing the ed25519 key type?

Reply:
I understand that, but we call `defaultParamUpdates` and then merge it with the existing updates. Isn't it unnecessary to declare the pubkeytypes here?

Reply:
Oh, updates are sparse.  Fields that are not changed are left nil.  This is a convention of cometbft, not me, but it does make sense.

Reply:
> Fields that are not changed are left nil, so if you don't want to change the Validator parameters, you leave that field nil

This is exactly my point. Why isn't PubKeyTypes nil?

Reply:
For this reason: https://github.com/kwilteam/kwil-db/pull/665#discussion_r1576491140

They should be nil, but if we change it now without defining a hardfork to do it, we will have consensus failures.